samples/wifi/promiscuous/src/main.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file
+ * @brief Wi-Fi Promiscuous sample
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(promiscuous, CONFIG_LOG_DEFAULT_LEVEL);
+
+#include "wifi_connection.h"
+#include <zephyr/net/ethernet.h>
+
+static int wifi_set_mode(int mode_val)
+{
+	struct net_if *iface = NULL;
+
+	iface = net_if_get_first_wifi();
+	if (iface == NULL) {
+		LOG_ERR("Failed to get Wi-Fi iface");
+		return -1;
+	}
+
+	if (net_eth_promisc_mode(iface, true)) {
+		LOG_ERR("Promiscuous mode enable failed");
+		return -1;
+	}
+
+	LOG_INF("Promiscuous mode enabled");
+	return 0;
+}
+
+int main(void)
+{
+	int ret;
+
+	if (wifi_set_mode(true)) {
+		return -1;
+	}
+
+	ret = try_wifi_connect();
```

Comment:
may be add a `TODO` to implement waiting for WPA supplicant to be managing the interface.